R/date-helpers.R

```diff
@@ -0,0 +1,41 @@
+# Type of day in terms of a number
+day_type <- function(day) {
+  dplyr::case_when(
+    day %in% c(1, 7) ~ 1,
+    day %in% 2:6 ~ 2
+  )
+}
+
+# Type of day, either Weekend or Weekday
+day_type_label <- function(day) {
+  dplyr::case_when(
+    day_type(day) == 1 ~ "Weekend",
+    day_type(day) == 2 ~ "Weekday"
```

Comment:
My thought for this was to also factor out repeated calls of the `day_type` function to the top of this function, then use the result from that call in the `case_when` block:
```suggestion
  day_type <- day_type(day)
  dplyr::case_when(
    day_type == 1 ~ "Weekend",
    day_type == 2 ~ "Weekday"
```
And the same for the `season_label` function below.

Reply:
My only reason against it is assigning the identifier from pointing at a function to pointing at an integer. The inlined version is more readable.

Reply:
My assumption is that fewer (repeated) function calls equates to more efficient code, but - like you - I've nothing to prove it. And the function being called is hardly complex...